book/routing.rst

```diff
@@ -808,6 +808,13 @@ Path     Parameters
 ``/es``  *won't match this route*
 =======  ========================
 
+.. tip::
+
+    The route requirements can also include container parameters, as explained
+    in :doc:`this article </cookbook/routing/service_container_parameters>`.
+    This comes in handy when the regular expression is very complex and used
+    time and again in your application.
```

Comment:
"time and again" is an idiom for "repeatedly". In the example of the UUID, the problem is that the regexp is horrible. But in other cases the regexp is simple ... but used in 100 different routes ... so it's convenient to turn it into a parameter. Should I reword this?